### DISCUSSION TITLE
A few Inconsistency pet peeves :-)

### DISCUSSION THREAD

**Discussion by walmsles** [aws/aws-pdk#611] [General]
Hi there.  I am new to this project (kind of).  I stumbled across it before the name change, and I like the concept of what you seem to be doing here. 

One thing that annoys me is the inconsistency of the typescript configuration for eslint/prettier between the project types:

- InfrastructureTsProject: uses " for strings
- Core PDK (base project) in ts uses " for strings
- AwsCdkConstructLibrary uses ' for strings.

is there a way to override this, or is there a standard that could be adopted across PDK generally to align the configuration of this tool?

My personal preference with typescript is imports and strings to be using ' (single quotes) :-) 

Some additional Questions:

1. Are you planning on publishing a public roadmap?
1. Do you have a chat place (other than here) that you gather to discuss this project (e.g. Discord)?
2. Is this project planned for some longevity?

Sorry for all the questions. I am interested in helping out - just trying to work out the how :-)

**Comment by walmsles** [comment]
Adding to this.  I have found how to override my quote preference in the base project using:

```typescript
const eslintJson = project.tryFindObjectFile('.eslintrc.json');
eslintJson?.addOverride('rules', {
  'prettier/prettier': ['error', { singleQuote: true }],
});
```

Can also apply this to the infrastructure project 😄

**Reply by walmsles** [reply]
But this is a  little messy IMO - would prefer if style was consistently enforced across PDK sub-projects.

**Comment by cogwirrel** [comment]
Hey! Sorry I just saw this! Have now updated my notification settings so that hopefully I see new discussions appear...

Glad to hear you've been playing with PDK since the name change! 😄 

`AwsCdkConstructLibrary` is unfortunately part of `projen` rather than PDK itself so we don't have control over its default style settings, but the idea is everything should be customisable.

Glad you found a way to update the styles! I think the neater way to do it would be to use `prettierOptions` in the constructor rather than the escape hatch, eg:

```ts
new InfrastructureTsProject({
  parent: monorepo,
  outdir: "packages/infra",
  name: "infra",
  typeSafeApi: api,
  prettierOptions: {
    settings: {
      singleQuote: true,
    },
  },
});
```

One cool thing about using PDK and projects as code is it means you can define common settings for things like styles in a single place - something I often do in my PDK-based projects is:

```ts
// Put all common typescript settings here
const commonTsProjectOptions = {
  prettierOptions: {
    settings: {
      singleQuote: true,
    },
  },
} satisfies Partial<TypeScriptProjectOptions>; // <- satisfies is handy for auto-complete while keeping it a const type

new InfrastructureTsProject({
  ...commonTsProjectOptions, // Spread them into each typescript project
  ...
});
```

Hope that helps! It might be possible to implement something where the monorepo applies certain consistent style settings to its child projects but that might get a bit confusing!

Re your other questions:

> 1. Are you planning on publishing a public roadmap?

I'm not sure about this one - @agdimech might be able to speak to this!

> 2. Do you have a chat place (other than here) that you gather to discuss this project (e.g. Discord)?

We don't right now! It would probably make sense to have a PDK channel in the [cdk.dev](https://cdk.dev/) slack - let me see if I can get in touch with the owner :)

> 3. Is this project planned for some longevity?

Yes! We definitely want to keep maintaining this long term. I work in a prototyping team where I build prototypes for AWS customers, and the majority of our prototypes are PDK based and are built with the intention of continuing to use PDK into production.

Sorry again for the slow reply! Certainly keen for your help! 😄

**Reply by walmsles** [reply]
No problem with the timeframe of your response.  This is open source, after all.  I agree using the projen props for prettier options is neater and is my preference, too. Will see if I can get that to work.  My issue was more making Visual Studio align to the linting rules, which involves eslint - the whole eslint/prettier thing appears to be quite frustrating (or maybe it's just me).  I will return to this - I like where it is going but have some other places to focus for a bit.

**Reply by cogwirrel** [reply]
Ahh gotcha, I think the right config files should hopefully get output if you have both `eslint: true` and `prettier: true` - though I’m never sure exactly what different IDEs need! Awesome! :)

**Reply by cogwirrel** [reply]
Quick update re (2): we now have a channel in the [cdk.dev](https://cdk.dev) slack!
